reana_commons/snakemake.py

```diff
@@ -169,9 +169,15 @@ def files(items):
     snakemake_validate(
         workflow_file=workflow_file, configfiles=configfiles, workdir=workdir
     )
-    snakemake_dag = _create_snakemake_dag(
-        workflow_file, configfiles=configfiles, **kwargs
-    )
+
+    # save the cwd to restore it after _create_snakemake_dag
```

Comment:
Cosmetics: Perhaps you can add to the comment the reason "... because this function may change cwd".

You can also enrich the docstrings of `_create_snakemake_dag()` in order to add a note about the working directory changing behaviour. (I guess it depends on whether the function received keyword argument `workdir`...)